src/keys/slip10.rs

```diff
@@ -7,79 +7,168 @@ use alloc::vec::Vec;
 use core::convert::TryFrom;
 
 use serde::{Deserialize, Serialize};
-use zeroize::{Zeroize, ZeroizeOnDrop, Zeroizing};
+use zeroize::{Zeroize, ZeroizeOnDrop};
 
 use crate::macs::hmac::HMAC_SHA512;
-#[cfg(feature = "ed25519")]
-use crate::signatures::ed25519;
-#[cfg(feature = "secp256k1")]
-use crate::signatures::secp256k1_ecdsa;
 
 // https://github.com/satoshilabs/slips/blob/master/slip-0010.md
 // https://github.com/bitcoin/bips/blob/master/bip-0032.mediawiki
 // https://en.bitcoin.it/wiki/BIP_0039
 
-#[derive(Clone, Copy, Debug, Eq, Hash, PartialEq, Serialize, Deserialize)]
-pub enum Curve {
-    #[cfg(feature = "ed25519")]
-    Ed25519,
-    #[cfg(feature = "secp256k1")]
-    Secp256k1,
+pub mod hazmat {
```

Comment:
Just curious, are these traits expected to be used externally?

Reply:
No, but they end up in public API so there's no way to make them private.